src/authDataManager/twitter.js

```diff
@@ -12,7 +12,7 @@ function validateAuthData(authData, options) {
   client.auth_token_secret = authData.auth_token_secret;
   
   return client.get("/1.1/account/verify_credentials.json").then((data) => {
-    if (data && data.id == authData.id) {
+    if (data && data.id_str == authData.id) {
```

Comment:
Given the twitter doc, where both id and id_str are provided, authData.id is not guaranteed to be neither a string nor a number. Given that, we should either type check, or use  `if (data && data.id_str == ''+authData.id) ` to make sure we're comparing the same types